# ANGLE needs to be build with /MD
diff --git build/common.gypi build/common.gypi
index 836df00b3..8bff1339e 100644
--- angle/build/common.gypi
+++ angle/build/common.gypi
@@ -236,7 +236,7 @@
                                 # Use the static C runtime to
                                 # match chromium and make sure
                                 # we don't depend on the dynamic
-                                'RuntimeLibrary': '0', # /MT (nondebug static)
+                                'RuntimeLibrary': '2', # /MD (nondebug dynamic)
                             }],
                         ],
                     },
