--- basekit/Makefile.lib	2014-09-08 18:20:23.000000000 +0200
+++ basekit/Makefile.lib	2014-08-04 12:01:00.000000000 +0200
@@ -58,7 +58,7 @@
 SYS ?= $(shell uname -s)
 
 ifeq ($(SYS),Darwin)
-CFLAGS += -falign-loops=16
+#CFLAGS += -falign-loops=16
 CFLAGS += -fPIC
 DLL_SUFFIX := dylib
 DLL_COMMAND := -dynamiclib
@@ -70,7 +70,7 @@
 endif
 
 ifeq ($(SYS),Linux)
-CFLAGS += -falign-loops=16
+#CFLAGS += -falign-loops=16
 CFLAGS += -fPIC
 endif
 
