--- giflib.org/lib/dgif_lib.c	2017-04-11 10:30:01 +0200
+++ giflib/lib/dgif_lib.c	2017-04-11 10:33:34 +0200
@@ -14,7 +14,9 @@
 #include <limits.h>
 #include <stdint.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdio.h>
 #include <string.h>
 
--- giflib.org/lib/gif_hash.h	2017-04-11 10:30:01 +0200
+++ giflib/lib/gif_hash.h	2017-04-11 10:34:05 +0200
@@ -10,7 +10,9 @@
 #ifndef _GIF_HASH_H_
 #define _GIF_HASH_H_
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdint.h>
 
 #define HT_SIZE			8192	   /* 12bits = 4096 or twice as big! */
--- giflib.org/lib/gif_hash.c	2017-04-11 10:30:01 +0200
+++ giflib/lib/gif_hash.c	2017-04-11 10:33:53 +0200
@@ -16,7 +16,9 @@
 * 14 Jun 89 - Version 1.0 by Gershon Elber.
 *****************************************************************************/
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdint.h>
 #include <fcntl.h>
 #include <stdio.h>
--- giflib.org/lib/egif_lib.c	2017-04-11 10:30:01 +0200
+++ giflib/lib/egif_lib.c	2017-04-11 10:33:43 +0200
@@ -11,7 +11,9 @@
  * 26 Jun 96 - Version 3.0 by Eric S. Raymond (Full GIF89 support)
  *****************************************************************************/
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdint.h>
 #include <stdlib.h>
 #include <stdio.h>
