--- http-parser.org/http_parser.gyp	2017-04-08 11:03:52 -0700
+++ http-parser/http_parser.gyp	2017-04-08 10:53:33 -0700
@@ -7,6 +7,7 @@
 {
   'target_defaults': {
     'default_configuration': 'Debug',
+    'msvs_configuration_platform': '<(platform)',
     'configurations': {
       # TODO: hoist these out and put them somewhere common, because
       #       RuntimeLibrary MUST MATCH across the entire project
@@ -15,7 +16,7 @@
         'cflags': [ '-Wall', '-Wextra', '-O0', '-g', '-ftrapv' ],
         'msvs_settings': {
           'VCCLCompilerTool': {
-            'RuntimeLibrary': 1, # static debug
+            'RuntimeLibrary': 2, # static debug
           },
         },
       },
@@ -24,7 +25,7 @@
         'cflags': [ '-Wall', '-Wextra', '-O3' ],
         'msvs_settings': {
           'VCCLCompilerTool': {
-            'RuntimeLibrary': 0, # static release
+            'RuntimeLibrary': 2, # static release
           },
         },
       }
