--- leveldb.org/Makefile	2017-03-07 23:06:08 -0800
+++ leveldb/Makefile	2017-03-08 00:14:34 -0800
@@ -25,10 +25,19 @@
 
 LDFLAGS += $(PLATFORM_LDFLAGS)
 LIBS += $(PLATFORM_LIBS)
+AR_FLAGS ?= -rs 
 
 LIBOBJECTS = $(SOURCES:.cc=.o)
 MEMENVOBJECTS = $(MEMENV_SOURCES:.cc=.o)
 
+ifeq ($(CC),cl)
+    OUT = -Fo:
+    CXXFLAGS += -nologo
+    CFLAGS += -nologo
+else
+    OUT = -o
+endif
+
 TESTUTIL = ./util/testutil.o
 TESTHARNESS = ./util/testharness.o $(TESTUTIL)
 
@@ -93,7 +102,7 @@
 endif
 
 $(SHARED3):
-	$(CXX) $(LDFLAGS) $(PLATFORM_SHARED_LDFLAGS)$(SHARED2) $(CXXFLAGS) $(PLATFORM_SHARED_CFLAGS) $(SOURCES) -o $(SHARED3) $(LIBS)
+	$(CXX) $(LDFLAGS) $(PLATFORM_SHARED_LDFLAGS)$(SHARED2) $(CXXFLAGS) $(PLATFORM_SHARED_CFLAGS) $(SOURCES) $(OUT)$(SHARED3) $(LIBS)
 
 endif  # PLATFORM_SHARED_EXT
 
@@ -105,10 +114,9 @@
 clean:
 	-rm -f $(PROGRAMS) $(BENCHMARKS) $(LIBRARY) $(SHARED) $(MEMENVLIBRARY) */*.o */*/*.o ios-x86/*/*.o ios-arm/*/*.o build_config.mk
 	-rm -rf ios-x86/* ios-arm/*
-
 $(LIBRARY): $(LIBOBJECTS)
 	rm -f $@
-	$(AR) -rs $@ $(LIBOBJECTS)
+	$(AR) $(AR_FLAGS)$@ $(LIBOBJECTS)
 
 db_bench: db/db_bench.o $(LIBOBJECTS) $(TESTUTIL)
 	$(CXX) $(LDFLAGS) db/db_bench.o $(LIBOBJECTS) $(TESTUTIL) -o $@ $(LIBS)
@@ -190,7 +198,7 @@
 
 $(MEMENVLIBRARY) : $(MEMENVOBJECTS)
 	rm -f $@
-	$(AR) -rs $@ $(MEMENVOBJECTS)
+	$(AR) $(AR_FLAGS)$@ $(MEMENVOBJECTS)
 
 memenv_test : helpers/memenv/memenv_test.o $(MEMENVLIBRARY) $(LIBRARY) $(TESTHARNESS)
 	$(CXX) $(LDFLAGS) helpers/memenv/memenv_test.o $(MEMENVLIBRARY) $(LIBRARY) $(TESTHARNESS) -o $@ $(LIBS)
@@ -220,8 +228,8 @@
 
 else
 .cc.o:
-	$(CXX) $(CXXFLAGS) -c $< -o $@
+	@$(CXX) $(CXXFLAGS) -c $< $(OUT)$@
 
 .c.o:
-	$(CC) $(CFLAGS) -c $< -o $@
+	@$(CC) $(CFLAGS) -c $< $(OUT)$@
 endif
--- leveldb.org/build_detect_platform	2017-03-07 23:06:08 -0800
+++ leveldb/build_detect_platform	2017-03-07 23:23:13 -0800
@@ -153,7 +153,7 @@
         PLATFORM_SHARED_CFLAGS=
         PLATFORM_SHARED_VERSIONED=
         ;;
-    OS_WINDOWS_CROSSCOMPILE | NATIVE_WINDOWS)
+    OS_WINDOWS_CROSSCOMPILE)
         PLATFORM=OS_WINDOWS
         COMMON_FLAGS="-fno-builtin-memcmp -D_REENTRANT -DOS_WINDOWS -DLEVELDB_PLATFORM_WINDOWS -DWINVER=0x0500 -D__USE_MINGW_ANSI_STDIO=1"
         PLATFORM_SOURCES="util/env_win.cc"
@@ -161,6 +161,15 @@
         PORT_FILE=port/port_win.cc
         CROSS_COMPILE=true
         ;;
+     NATIVE_WINDOWS)
+        PLATFORM=OS_WINDOWS
+        COMMON_FLAGS="-D_REENTRANT -DOS_WINDOWS -DLEVELDB_PLATFORM_WINDOWS -DWINVER=0x0500 -D__USE_MINGW_ANSI_STDIO=1"
+        PLATFORM_SOURCES="util/env_win.cc"
+        PLATFORM_LIBS="-lshlwapi"
+        PLATFORM_SHARED_LDFLAGS=
+        PORT_FILE=port/port_win.cc
+        CROSS_COMPILE=true
+        ;;
     *)
         echo "Unknown platform!" >&2
         exit 1
--- leveldb.org/db/c.cc	2017-03-07 23:06:08 -0800
+++ leveldb/db/c.cc	2017-03-07 23:23:13 -0800
@@ -5,7 +5,9 @@
 #include "leveldb/c.h"
 
 #include <stdlib.h>
+#ifndef _MSC_VER
 #include <unistd.h>
+#endif
 #include "leveldb/cache.h"
 #include "leveldb/comparator.h"
 #include "leveldb/db.h"
--- leveldb.org/db/db_iter.cc	2017-03-07 23:06:08 -0800
+++ leveldb/db/db_iter.cc	2017-03-07 23:23:13 -0800
@@ -14,6 +14,11 @@
 #include "util/mutexlock.h"
 #include "util/random.h"
 
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
 namespace leveldb {
 
 #if 0
--- leveldb.org/port/port_win.h	2017-03-07 23:06:08 -0800
+++ leveldb/port/port_win.h	2017-03-07 23:23:13 -0800
@@ -32,7 +32,7 @@
 #define STORAGE_LEVELDB_PORT_PORT_WIN_H_
 
 #ifdef _MSC_VER
-#define snprintf _snprintf
+//#define snprintf _snprintf
 #define close _close
 #define fread_unlocked _fread_nolock
 #endif
