--- libcoroutine/Makefile.lib	2009-10-20 21:20:10.000000000 +0200
+++ libcoroutine.old/Makefile.lib	2013-03-12 10:48:38.626672271 +0100
@@ -51,7 +51,7 @@
 CFLAGS = $(OPTIMIZE) $(WARN) $(HEADERS) #--param max-inline-insns-single=500
 
 # Uncommment for Coros to register their stack with Valgrind
-#CFLAGS += -DUSE_VALGRIND
+CFLAGS += -DUSE_VALGRIND
 
 ### PLATFORM #####################################################
 
