--- libcoroutine.org/Makefile	2017-03-10 12:44:13 +0100
+++ libcoroutine/Makefile	2017-03-10 15:57:02 +0100
@@ -3,5 +3,5 @@
 
 # Manually control which coro implementation to use
 #CFLAGS += -DUSE_UCONTEXT   	# preferred on OSX, Linux and friends
-#CFLAGS += -DUSE_FIBERS		# preferred on Windows
+CFLAGS += -DUSE_FIBERS		# preferred on Windows
 #CFLAGS += -DUSE_SETJMP		# method of last resort
--- libcoroutine.org/Makefile.lib	2017-03-10 12:44:13 +0100
+++ libcoroutine/Makefile.lib	2017-03-10 16:38:08 +0100
@@ -38,7 +38,7 @@
 HEADERS  :=-I. -I./source
 others := $(dependson)
 others := $(addprefix -I../,$(others))
-others := $(addsuffix /_build/headers,$(others))
+others := $(addsuffix /source,$(others))
 HEADERS += $(others)
 
 DLL_L := $(dependson)
@@ -58,7 +58,7 @@
 SYS ?= $(shell uname -s)
 
 ifeq ($(SYS),Darwin)
-CFLAGS += -falign-loops=16
+#CFLAGS += -falign-loops=16
 CFLAGS += -fPIC
 DLL_SUFFIX := dylib
 DLL_COMMAND := -dynamiclib
@@ -70,7 +70,7 @@
 endif
 
 ifeq ($(SYS),Linux)
-CFLAGS += -falign-loops=16
+#CFLAGS += -falign-loops=16
 CFLAGS += -fPIC
 endif
 
@@ -132,7 +132,7 @@
 
 DLL_L := $(dependson)
 DLL_L := $(addprefix $(LINKDIRFLAG)../,$(DLL_L))
-DLL_L := $(addsuffix /_build/dll,$(DLL_L))
+DLL_L := $(addsuffix /_build/lib/,$(DLL_L))
 
 DLL_l := $(dependson)
 DLL_l := $(addprefix $(LINKLIBFLAG),$(DLL_l))
@@ -190,8 +190,12 @@
 
 $(LIBR): $(objects)
 ifneq ($(asmfiles),)
+ifneq (,$(findstring Windows,$(SYS)))
+	yasm $(CFLAGS) $(asmfiles) -o _build/objs/asm.o || true
+else
 	$(CC) $(CFLAGS) -c $(asmfiles) $(CCOUTFLAG)_build/objs/asm.o || true
 endif
+endif
 	$(AR) $(ARFLAGS) $(AROUTFLAG)$@ _build/objs/*.o
 	$(RANLIB) $@
 
--- libcoroutine.org/source/taskimpl.h	2017-03-10 15:52:19 +0100
+++ libcoroutine/source/taskimpl.h	2017-03-10 16:34:34 +0100
@@ -31,19 +31,21 @@
 
 #include <errno.h>
 #include <stdlib.h>
-#include <unistd.h>
 #include <string.h>
 #include <assert.h>
 #include <time.h>
-#include <sys/time.h>
 #include <sys/types.h>
+#ifndef _MSC_VER
+#include <unistd.h>
+#include <sys/time.h>
 #include <sys/wait.h>
 #include <sched.h>
+#include <sys/utsname.h>
+#endif
 #include <signal.h>
 #if USE_UCONTEXT
 #include <ucontext.h>
 #endif
-#include <sys/utsname.h>
 #include <inttypes.h>
 //#include "task.h"
 
