--- SDL2/src/audio/xaudio2/SDL_xaudio2.c.org	2017-02-26 00:18:45 -0800
+++ SDL2/src/audio/xaudio2/SDL_xaudio2.c	2017-02-26 00:19:10 -0800
@@ -65,7 +65,7 @@
    The XAudio2 implementation available in the Windows 8 SDK targets Windows 8 and newer.
    If you want to build SDL with XAudio2 support you should install the DirectX SDK.
  */
-#include <dxsdkver.h>
+//#include <dxsdkver.h>
 #if (!defined(_DXSDK_BUILD_MAJOR) || (_DXSDK_BUILD_MAJOR < 1284))
 #  pragma message("Your DirectX SDK is too old. Disabling XAudio2 support.")
 #else
