diff --git skia/include/gpu/gl/GrGLFunctions.h skia/include/gpu/gl/GrGLFunctions.h
index 771e1ebcec..3e6f6077c6 100644
--- skia/include/gpu/gl/GrGLFunctions.h
+++ skia/include/gpu/gl/GrGLFunctions.h
@@ -186,6 +186,20 @@ typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLVertexAttribDivisorProc)(GrGLuint ind
 typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLVertexAttribIPointerProc)(GrGLuint indx, GrGLint size, GrGLenum type, GrGLsizei stride, const GrGLvoid* ptr);
 typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLVertexAttribPointerProc)(GrGLuint indx, GrGLint size, GrGLenum type, GrGLboolean normalized, GrGLsizei stride, const GrGLvoid* ptr);
 typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLViewportProc)(GrGLint x, GrGLint y, GrGLsizei width, GrGLsizei height);
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLVertexAttrib2fProc)(GrGLuint indx, GrGLfloat x, GrGLfloat y);
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLVertexAttrib3fProc)(GrGLuint indx, GrGLfloat x, GrGLfloat y, GrGLfloat z);
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLVertexAttrib4fProc)(GrGLuint indx, GrGLfloat x, GrGLfloat y, GrGLfloat z, GrGLfloat w);
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLVertexAttrib1fvProc)(GrGLuint indx, const GrGLfloat *value);
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLBlendEquationProc)(GrGLenum mode);
+// extention?
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLBlendEquationSeparateProc)(GrGLenum modeRGB, GrGLenum modeAlpha);
+// extention?
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLBlendFuncSeparateProc)(GrGLenum sfactorRGB, GrGLenum dfactorRGB, GrGLenum sfactorAlpha, GrGLenum dfactorAlpha);
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLClearDepthProc)(GrGLclampf depth);
+typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLDepthFuncProc)(GrGLenum func);
+typedef GrGLint (GR_GL_FUNCTION_TYPE* GrGLGetAttribLocationProc)(GrGLuint program, const GrGLchar *name);
+typedef GrGLint (GR_GL_FUNCTION_TYPE* GrGLGetFloatvProc)(GrGLenum pname, GrGLfloat *params);
+typedef GrGLint (GR_GL_FUNCTION_TYPE* GrGLGetBooleanvProc)(GrGLenum pname, GrGLboolean *params);
 
 /* GL_NV_path_rendering */
 typedef GrGLvoid (GR_GL_FUNCTION_TYPE* GrGLMatrixLoadfProc)(GrGLenum matrixMode, const GrGLfloat* m);
diff --git skia/include/gpu/gl/GrGLInterface.h skia/include/gpu/gl/GrGLInterface.h
index 3afa6be9fd..d4e2fdaf68 100644
--- skia/include/gpu/gl/GrGLInterface.h
+++ skia/include/gpu/gl/GrGLInterface.h
@@ -296,6 +296,17 @@ public:
         GrGLFunction<GrGLVertexAttribIPointerProc> fVertexAttribIPointer;
         GrGLFunction<GrGLVertexAttribPointerProc> fVertexAttribPointer;
         GrGLFunction<GrGLViewportProc> fViewport;
+        GrGLFunction<GrGLVertexAttrib2fProc> fVertexAttrib2f;
+        GrGLFunction<GrGLVertexAttrib3fProc> fVertexAttrib3f;
+        GrGLFunction<GrGLVertexAttrib4fProc> fVertexAttrib4f;
+        GrGLFunction<GrGLVertexAttrib1fvProc> fVertexAttrib1fv;
+        GrGLFunction<GrGLBlendEquationSeparateProc> fBlendEquationSeparate;
+        GrGLFunction<GrGLBlendFuncSeparateProc> fBlendFuncSeparate;
+        GrGLFunction<GrGLClearDepthProc> fClearDepth;
+        GrGLFunction<GrGLDepthFuncProc> fDepthFunc;
+        GrGLFunction<GrGLGetAttribLocationProc> fGetAttribLocation;
+        GrGLFunction<GrGLGetFloatvProc> fGetFloatv;
+        GrGLFunction<GrGLGetBooleanvProc> fGetBooleanv;
 
         /* GL_NV_path_rendering */
         GrGLFunction<GrGLMatrixLoadfProc> fMatrixLoadf;
diff --git skia/src/gpu/gl/GrGLAssembleInterface.cpp skia/src/gpu/gl/GrGLAssembleInterface.cpp
index 1c10214382..6e79bc1452 100644
--- skia/src/gpu/gl/GrGLAssembleInterface.cpp
+++ skia/src/gpu/gl/GrGLAssembleInterface.cpp
@@ -266,6 +266,18 @@ const GrGLInterface* GrGLAssembleGLInterface(void* ctx, GrGLGetProc get) {
 
     GET_PROC(VertexAttribPointer);
     GET_PROC(Viewport);
+    GET_PROC(VertexAttrib2f);
+    GET_PROC(VertexAttrib3f);
+    GET_PROC(VertexAttrib4f);
+    GET_PROC(VertexAttrib1fv);
+    GET_PROC(BlendEquation);
+    GET_PROC(BlendEquationSeparate);
+    GET_PROC(BlendFuncSeparate);
+    GET_PROC(ClearDepth);
+    GET_PROC(DepthFunc);
+    GET_PROC(GetAttribLocation);
+    GET_PROC(GetFloatv);
+    GET_PROC(GetBooleanv);
     GET_PROC(BindFragDataLocationIndexed);
 
     if (glVer >= GR_GL_VER(3,0) || extensions.has("GL_ARB_vertex_array_object")) {
@@ -749,6 +761,18 @@ const GrGLInterface* GrGLAssembleGLESInterface(void* ctx, GrGLGetProc get) {
 
     GET_PROC(VertexAttribPointer);
     GET_PROC(Viewport);
+    GET_PROC(VertexAttrib2f);
+    GET_PROC(VertexAttrib3f);
+    GET_PROC(VertexAttrib4f);
+    GET_PROC(VertexAttrib1fv);
+    GET_PROC(BlendEquation);
+    GET_PROC(BlendEquationSeparate);
+    GET_PROC(BlendFuncSeparate);
+    GET_PROC_SUFFIX(ClearDepth, f);
+    GET_PROC(DepthFunc);
+    GET_PROC(GetAttribLocation);
+    GET_PROC(GetFloatv);
+    GET_PROC(GetBooleanv);
     GET_PROC(BindFramebuffer);
     GET_PROC(BindRenderbuffer);
     GET_PROC(CheckFramebufferStatus);
