--- zita-resampler.org/libs/Makefile	2012-10-26 22:58:55.000000000 +0200
+++ zita-resampler/libs/Makefile	2017-04-13 18:23:14.456841988 +0200
@@ -28,11 +28,13 @@
 MINVERS = 3.0
 VERSION = $(MAJVERS).$(MINVERS)
 DISTDIR = zita-resampler-$(VERSION)
-
+CC ?= g++
 
 CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CXXFLAGS += -Wall -fPIC -O2 -ffast-math
-CXXFLAGS += -march=native
+ifeq ($(CC),gcc)
+	CXXFLAGS += -march=native
+endif
 LDFLAGS += 
 LDLIBS +=
 
@@ -45,8 +47,10 @@
 ZITA-RESAMPLER_H = zita-resampler/resampler.h zita-resampler/resampler-table.h zita-resampler/vresampler.h
 
 
+libzita-resampler.a : $(ZITA-RESAMPLER_O)
+	ar rcs $@ $(ZITA-RESAMPLER_O)
 $(ZITA-RESAMPLER_MIN): $(ZITA-RESAMPLER_O)
-	g++ -shared $(LDFLAGS) -Wl,-soname,$(ZITA-RESAMPLER_MAJ) -o $(ZITA-RESAMPLER_MIN) $(ZITA-RESAMPLER_O) $(ZITA-RESAMPLER_DEP)
+	$(CC) -shared $(LDFLAGS) -Wl,-soname,$(ZITA-RESAMPLER_MAJ) -o $(ZITA-RESAMPLER_MIN) $(ZITA-RESAMPLER_O) $(ZITA-RESAMPLER_DEP)
 
 $(ZITA-RESAMPLER_O):	$(ZITA-RESAMPLER_H)
 
