diff -rupN skia-origin/src/core/SkPath.cpp skia/src/core/SkPath.cpp
--- skia-origin/src/core/SkPath.cpp	2013-02-28 22:17:34.000000000 +0100
+++ skia/src/core/SkPath.cpp	2013-04-27 22:45:43.000000000 +0200
@@ -1387,10 +1387,11 @@ void SkPath::addPath(const SkPath& path,
     this->addPath(path, matrix);
 }
 
-void SkPath::addPath(const SkPath& path, const SkMatrix& matrix) {
+void SkPath::addPath(const SkPath& path, const SkMatrix& matrix, bool skipMoveTo) {
     SkPathRef::Editor(&fPathRef, path.countVerbs(), path.countPoints());
 
     fIsOval = false;
+    int i = 0;
 
     RawIter iter(path);
     SkPoint pts[4];
@@ -1398,9 +1399,10 @@ void SkPath::addPath(const SkPath& path,
 
     SkMatrix::MapPtsProc proc = matrix.getMapPtsProc();
 
-    while ((verb = iter.next(pts)) != kDone_Verb) {
+    while ((verb = iter.next(pts)) != kDone_Verb && ++i) {
         switch (verb) {
             case kMove_Verb:
+                if (i == 1 && skipMoveTo) break;
                 proc(matrix, &pts[0], &pts[0], 1);
                 this->moveTo(pts[0]);
                 break;
diff -rupN skia-origin/include/core/SkPath.h skia/include/core/SkPath.h
--- skia-origin/include/core/SkPath.h	2013-02-28 22:17:31.000000000 +0100
+++ skia/include/core/SkPath.h	2013-04-27 22:35:30.000000000 +0200
@@ -708,7 +708,7 @@ public:
     /** Add a copy of src to the path, transformed by matrix
         @param src  The path to add as a new contour
     */
-    void addPath(const SkPath& src, const SkMatrix& matrix);
+    void addPath(const SkPath& src, const SkMatrix& matrix, bool skipMoveTo=false);
 
     /**
      *  Same as addPath(), but reverses the src input
