Index: pylib/gyp/input.py
===================================================================
--- gyp/pylib/gyp/input.py	(révision 1967)
+++ gyp/pylib/gyp/input.py	(copie de travail)
@@ -62,13 +62,15 @@
   # considered PathSections. Using manual string matching since that
   # is much faster than the regexp and this can be called hundreds of
   # thousands of times so micro performance matters.
-  if "_" in section:
-    tail = section[-6:]
+  last_underscore = section.rfind("_")
+  if last_underscore != -1:
+    tail = section[last_underscore:]
+    if tail[-1] == '%':
+      tail = tail[:-1]
     if tail[-1] == 's':
       tail = tail[:-1]
-    if tail[-5:] in ('_file', '_path'):
+    if tail in ('_file', '_path', '_dir'):
       return True
-    return tail[-4:] == '_dir'
 
   return False
 
Index: gyp_main.py
===================================================================
--- gyp/gyp_main.py	(révision 1967)
+++ gyp/gyp_main.py	(copie de travail)
@@ -15,4 +15,4 @@
   import gyp
 
 if __name__ == '__main__':
-  sys.exit(gyp.script_main())
+    sys.exit(gyp.main(sys.argv[1:]))
