--- libcoroutine/Makefile.lib	2009-10-20 21:20:10.000000000 +0200
+++ libcoroutine.old/Makefile.lib	2013-03-12 10:48:38.626672271 +0100
@@ -132,7 +132,7 @@
 
 DLL_L := $(dependson)
 DLL_L := $(addprefix $(LINKDIRFLAG)../,$(DLL_L))
-DLL_L := $(addsuffix /_build/dll,$(DLL_L))
+DLL_L := $(addsuffix /_build/lib/,$(DLL_L))
 
 DLL_l := $(dependson)
 DLL_l := $(addprefix $(LINKLIBFLAG),$(DLL_l))

--- libcoroutine/Makefile	2013-03-12 13:46:26.943058532 +0100
+++ libcoroutine.old/Makefile	2013-03-12 09:38:59.374520958 +0100
@@ -2,6 +2,6 @@
 CFLAGS += -DBUILDING_CORO_DLL $(IOVMALLFLAGS)
 
 # Manually control which coro implementation to use
-#CFLAGS += -DUSE_UCONTEXT  	# preferred on OSX, Linux and friends
+CFLAGS += -DUSE_UCONTEXT   	# preferred on OSX, Linux and friends
 #CFLAGS += -DUSE_FIBERS		# preferred on Windows
 #CFLAGS += -DUSE_SETJMP		# method of last resort
