--- libcoroutine/Makefile	2013-03-12 21:03:34.649354376 +0100
+++ libcoroutine/Makefile-patch	2013-03-12 21:04:00.165354988 +0100
@@ -2,6 +2,6 @@
 CFLAGS += -DBUILDING_CORO_DLL $(IOVMALLFLAGS)
 
 # Manually control which coro implementation to use
-#CFLAGS += -DUSE_UCONTEXT   	# preferred on OSX, Linux and friends
+CFLAGS += -DUSE_UCONTEXT   	# preferred on OSX, Linux and friends
 #CFLAGS += -DUSE_FIBERS		# preferred on Windows
 #CFLAGS += -DUSE_SETJMP		# method of last resort
