--- zita-resampler/libs/Makefile	2012-10-26 22:58:55.000000000 +0200
+++ ../zita-resampler/libs/Makefile	2012-12-10 15:31:44.864714778 +0100
@@ -45,6 +45,8 @@
 ZITA-RESAMPLER_H = zita-resampler/resampler.h zita-resampler/resampler-table.h zita-resampler/vresampler.h
 
 
+libzita-resampler.a : $(ZITA-RESAMPLER_O)
+	ar rcs $@ $(ZITA-RESAMPLER_O)
 $(ZITA-RESAMPLER_MIN): $(ZITA-RESAMPLER_O)
 	g++ -shared $(LDFLAGS) -Wl,-soname,$(ZITA-RESAMPLER_MAJ) -o $(ZITA-RESAMPLER_MIN) $(ZITA-RESAMPLER_O) $(ZITA-RESAMPLER_DEP)
 
